src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int NUM_OF_CLASSES = 5;
+    public static final int DEFAULT_RADIUS = 10;
+    public static final int MAX_SIZE = 100;
+
+    public Figure getRandomFigure() {
+        switch (new Random().nextInt(NUM_OF_CLASSES)) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomIsoscelesTrapezoid();
+            case 2:
+                return getRandomRectangle();
+            case 3:
+                return getRandomRightTriangle();
+            case 4:
+                return getRandomSquare();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_RADIUS, Colors.WHITE);
+    }
+
+    public Figure getRandomCircle() {
+        return new Circle(new Random().nextInt(MAX_SIZE),
+                Colors.valueOf(new ColorSupplier().getRandomColor()));
```

Comment:
You are creating a lot of `Random` and `ColorSupplier` objects in each method. Can we optimize it to create them only once? [Reference](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#think-about-which-variables-should-be-local-in-the-method-and-which-should-be-class-level).